validator/keymanager/remote-web3signer/client_test.go

```diff
@@ -0,0 +1,109 @@
+package remote_web3signer
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// MockClient is the mock client
+type mockClient struct {
+	DoFunc func(req *http.Request) (*http.Response, error)
+}
+
+var (
+	// GetDoFunc fetches the mock client's `Do` func
+	GetDoFunc func(req *http.Request) (*http.Response, error)
+)
+
+// Do is the mock client's `Do` func
+func (*mockClient) Do(req *http.Request) (*http.Response, error) {
+	return GetDoFunc(req)
+}
+
+func TestClient_Sign_HappyPath(t *testing.T) {
+	json := `{
+  		"signature": "0xb3baa751d0a9132cfe93e4e3d5ff9075111100e3789dca219ade5a24d27e19d16b3353149da1833e9b691bb38634e8dc04469be7032132906c927d7e1a49b414730612877bc6b2810c8f202daf793d1ab0d6b5cb21d52f9e52e883859887a5d9"
+	}`
+	// create a new reader with that JSON
+	r := ioutil.NopCloser(bytes.NewReader([]byte(json)))
+	GetDoFunc = func(*http.Request) (*http.Response, error) {
```

Comment:
Implementing these tests with a RoundTripper should be more flexible than replacing a package level variable for each test case. For example if you wanted to expose more of the http.Client methodset, or implement table-driven tests.